redaxo/src/addons/backup/vendor/tar.php

```diff
@@ -106,6 +106,7 @@ protected function __computeUnsignedChecksum($bytestring) {
   // PRIVATE ACCESS FUNCTION
   protected function __parseNullPaddedString($string) {
     $position = strpos($string,chr(0));
+    assert(is_int($position));
```

Comment:
Nach meinem Verständnis dürfen wir assert() nur verwenden wenn wir sicher sind das der zustand nie in production auftritt... in dev kommt es zum error... in prod wird der assert() nicht geprüft..

Passt hier dann nicht, oder?

Reply:
aus meiner sicht eignet sich assert auch für stellen, wo man meint, dass es nie vorkommen sollte, aber sich nicht ganz sicher ist.
Denn an solchen Stellen hat es eben den Vorteil, dass in prod es weiterhin so funktioniert wie vorher.

Und hier im konkreten Fall bin ich mir eben nicht sicher. Eigentlich denke ich, dass die tar klasse da nur Strings übergeben sollte, die ein Null-Byte enthalten. Aber ich kann auch nicht ganz ausschließen, dass nicht doch andere übergeben werden, und es dann mit `false` trotzdem weiter läuft.

Reply:
Wenn aus `substr($string,0,false)` was sinnvolles rauskommt 👍

Reply:
Naja, false wird zu 0, somit kommt da immer ein leerer String raus. Bedingt sinnvoll, aber eben für mich auch nicht völlig ausgeschlossen, dass die tar klasse damit arbeitet.